### PR TITLE
Added model sampling settngs functionaility

### LINKED ISSUE #47: [feature]: Add model sampling settings
- [ ] **Context Length**: `n_ctx`
- [ ] **Batch Size**: `n_batch`
- [ ] **N Predict**: `nPredict`
- [ ] **Temperature**: `temp`

### PR DESCRIPTION
**Pull request by Murkeee**
Created new component ```ChatSettings.jsx``` and corresponding css ```chatSettings.css```
Modified files ```ChatPage.jsx``` and ```worker.js```
Added media files ```left-arrow.svg``` and ```settings-icon.svg```

-Added a togglable settings menu on the left side of the chat page
-Added sliders and number inputs for ```n_batch, n_ctx, n_predict, temp```
-Added button to reload the model
-Added functionality to update the completion parameters ```n_predict``` and ```temp```
-Added functionailty to update the model parameters ```n_batch and n_ctx```

closes #47

### CODE REVIEW

File: src/utils/worker.js [hunk 1]
```diff
@@ -17,14 +17,13 @@ const CONFIG_PATHS = {
 
 const DEFAULT_CHAT_TEMPLATE = "{% for message in messages %}{{'<|im_start|>' + message['role'] + '\n' + message['content'] + '<|im_end|>' + '\n'}}{% endfor %}{% if add_generation_prompt %}{{ '<|im_start|>assistant\n' }}{% endif %}";
 
-const engines = {
+let engines = {
```
Comment (cbh778899) [inline src/utils/worker.js:20]: If you are not changing the whole object, you can keep it const, unless it's freeze()'ed, you can do whatever change inside that.

File: src/components/chat/ChatSettings.jsx [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+              max={maximums.nBatch}
+              value={nBatch}
+              onChange={(e) => nBatchValue(e.target.value)}/>
+          </div>
+          <b  className='notice'>Changing the context length or Batch size will require you to reload the model.</b>
+          <hr/>
+          <button onClick={reload} className='reload'>Reload Model</button>
+        </div>
+        <button onClick={openSettings} className={`toggle ${open ? 'open' : ''}`}>
+          {open ? <img className="settingsIcon" src={settingsIcon}></img> : <img className='leftArrow' src={leftArrow}></img>}
```
[the remark this answers is not in the document]
Comment (cbh778899) [reply src/components/chat/ChatSettings.jsx:114]: For more icons please see [https://icons.getbootstrap.com/](https://icons.getbootstrap.com/), after you found the icon you want to use, just import the name in camel case. This is a convenient library and I personally use it a lot.

[the remark this answers is not in the document]
Comment (Murkeee) [reply src/components/chat/ChatSettings.jsx]: Ahhhh i did not know this was the case. Thank you this is super useful.

File: src/utils/worker.js (hunk 1 above)
Comment (Murkeee) [reply src/utils/worker.js]: I changed it to let before for earlier testing and forgot to change back. Fixed.

File: src/components/chat/ChatPage.jsx [hunk 3]
```diff
@@ -33,7 +36,7 @@ export default function ChatPage() {
                 toggleDoingInference(false);
                 setPendingContent('');
             }
-        })
+        }, parseInt(nPredict), parseFloat(temp))
```
Comment (cbh778899) [inline src/components/chat/ChatPage.jsx:39]: This is not a problem, just a hint - In js if you are not strict on int/float, just want it to be a number, you can just do it like `+nPredict` and `+temp`, so the code is simpler.

**Review by cbh778899 — APPROVED** [review]
LGTM

**Pull request merged**